tests/integration_test/dns_resolver_test.go

```diff
@@ -25,63 +25,85 @@ import (
 	"github.com/GoogleCloudPlatform/esp-v2/tests/env/platform"
 )
 
-func fullDns(dns string) string {
+func toFqdnWithRoot(dns string) string {
 	return dns + "."
 }
 
 func TestDnsResolver(t *testing.T) {
 	t.Parallel()
-	s := env.NewTestEnv(comp.TestDnsResolver, platform.EchoSidecar)
-
-	// Setup dns resolver.
-	backendHost := "dns-resolver-test-backend"
-	localHost := "127.0.0.1"
-	dnsRecords := map[string]string{
-		fullDns(backendHost): localHost,
-	}
-	dnsResolver := comp.NewDnsResolver(s.Ports().DnsResolverPort, dnsRecords)
-	go func() {
-		if err := dnsResolver.ListenAndServe(); err != nil {
-			t.Fatalf("Failed to set udp listener %s\n", err.Error())
-		}
-	}()
-
-	// Check dns resolver's health.
-	dnsResolverAddress := fmt.Sprintf("127.0.0.1:%v", s.Ports().DnsResolverPort)
-	if err := comp.CheckDnsResolverHealth(dnsResolverAddress, backendHost, localHost); err != nil {
-		t.Fatalf("DNS Resolver is not healthy: %v", err)
-	}
-
-	// Setup the whole test framework.
-	s.SetBackendAddress(fmt.Sprintf("http://%s:%v", backendHost, s.Ports().BackendServerPort))
-	args := []string{"--service_config_id=test-config-id",
-		"--rollout_strategy=fixed", "--healthz=/healthz", "--dns_resolver_addresses=" + dnsResolverAddress}
 
 	testCase := []struct {
 		desc            string
+		backendHost     string
 		isResolveFailed bool
 		wantResp        string
 		wantError       string
 	}{
 		{
-			desc:     "resolve domain name successfully",
-			wantResp: `{"message":"hello"}`,
+			desc:        "resolve non-FQDN domain name successfully",
+			backendHost: "dns-resolver-test-backend",
```

Comment:
For PQDN, it will append domain to the host. Can we get the domain in the test and put that in dns_resolver?

Like if we can somehow get `corp.google.com` is local domain, we put `dns-resolver-test-backend.corp.google.com` in dns_resolver and let envoy to discover it. Current case is not obvious.

Reply:
We can, but that's not what we should be testing, right? We should ensure that Envoy will make calls to discover the PQDN instead of the FQDN. The other tests with FQDN test the case you are describing.

Reply:
IIUC, all the dns requests sent from envoy are all FQDN by appending root dot to the full dns guess. I was trying to say whether we can test the process of envoy trying all the possible FQDN by appending domains like
- `dns-resolver-test-backend.`
- `dns-resolver-test-backend.XXX`
- `dns-resolver-test-backend.YYY`
-  `dns-resolver-test-backend.test-instance-domain.com.`
...

It can be achieved by setting `dns-resolver-test-backend` as backend and `dns-resolver-test-backend.test-instance-domain.com.`(we need know the domain of machine running the test) in the dns resolver. The backend host in other test cases look already have the full name so they don't cover this case.

Reply:
I don't see the benefit of testing this, this is part of the DNS spec. As long as both sides are using a spec-compliant DNS resolver, it should be fine.

Also based on my testing, it doesn't just use the host name. Here are some example queries from Envoy on my workstation:
```
;dns-resolver-test-backend.prod.google.com
;dns-resolver-test-backend.prodz.google.com.
;dns-resolver-test-backend.google.com.
;dns-resolver-test-backend.svl.corp.google.com.
;dns-resolver-test-backend.
```

I don't know where it gets the first 2 from...

Reply:
In this case, this testCase has no difference with other testCase. We are expecting resolving `dns-resolver-test-backend.`, which is already a FQDN.

Reply:
We spoke offline and decided it was worth testing it, since envoy seems to make different requests based on if it has the host name.